src/ApplicationSampler.hpp

```diff
@@ -67,8 +67,10 @@ namespace geopm
             virtual void time_zero(const geopm_time_s &start_time) = 0;
             /// @brief Update the record buffer by clearing out old
             ///        records and providing a new cache for
-            ///        subsequent calls to the get_records() method.
-            virtual void update_records(void) = 0;
+            ///        subsequent calls to the get_records()
+            ///        method. Also update cache of application
+            ///        status for use by hint and progress APIs.
+            virtual void update(const geopm_time_s &curr_time) = 0;
```

Comment:
document curr_time

Reply:
this is exposed for testing?

Reply:
Yes...might be nice to have a Clock singleton that we could mock for time values.